src/views/MaintenanceView.vue

```diff
@@ -0,0 +1,389 @@
+ */
+const processUpdate = (update: InstallValidation) => {
+  // TODO: if (update.state === 'installed')
+  validationState.value = update
+
+  // Update each task state
+  for (const task of tasks.value) {
+    task.loading = update[task.id] === undefined
+    // Mark resolved
+    if (task.state === 'error' && update[task.id] === 'OK') task.resolved = true
+    if (update[task.id]) task.state = update[task.id]
+  }
+
+  // Final update
+  if (!update.inProgress && isRefreshing.value) {
+    isRefreshing.value = false
+    for (const task of tasks.value) {
+      task.state = update[task.id] ?? 'skipped'
+      task.loading = false
+    }
+  }
+}
+
+/** Cherry-picked properties for {@link createTask} function. */
+type TaskProps = Pick<
+  MaintenanceTask,
+  'id' | 'headerImg' | 'onClick' | 'requireConfirm' | 'severity'
+> & {
+  /** Icon to show on the button, e.g. `'pi pi-external-link'` */
+  buttonIcon?: string
+}
+
+/**
+ * Helper function that creates a maintenance task to display.
+ * @param properties Task properties that are not stored in i18n files
+ */
+function createTask({
+  id,
+  headerImg,
+  severity,
+  requireConfirm,
+  buttonIcon,
+  onClick
+}: TaskProps): MaintenanceTask {
+  return {
+    id,
+    name: t(`maintenance.tasks.${id}.name`),
+    state: null,
+    description: st(`maintenance.tasks.${id}.description`, undefined),
+    descriptionOk: st(`maintenance.tasks.${id}.descriptionOk`, undefined),
+    errorDescription: st(`maintenance.tasks.${id}.errorDescription`, undefined),
+    detail: st(`maintenance.tasks.${id}.detail`, undefined),
+    confirmText: st(`maintenance.tasks.${id}.confirmText`, undefined),
+    headerImg,
+    severity,
+    requireConfirm,
+    onClick,
+    loading: minDurationRef(true, minRefreshTime),
+    button: {
+      text: t(`maintenance.tasks.${id}.buttonText`),
+      icon: buttonIcon
+    }
+  }
+}
+
+const installRequirements = async () => {
+  terminalVisible.value = true
+  try {
+    await electron.uv.installRequirements()
+  } catch (error) {
+    toast.add({
+      severity: 'error',
+      summary: 'Failed to install requirements',
+      detail: error.message,
+      life: 10_000
+    })
+  }
+  await completeValidation(false)
+  if (tasks.value.find((x) => x.id === 'pythonPackages')?.state === 'OK') {
+    terminalVisible.value = false
+  }
+}
+
+const resetVenv = async () => {
+  terminalVisible.value = true
+  const success = await electron.uv.resetVenv()
+  if (!success) return
+
+  terminalVisible.value = false
+  await completeValidation(false)
+}
+
+const clearCache = async () => {
+  terminalVisible.value = true
+  const success = await electron.uv.clearCache()
+  if (!success) return
+
+  terminalVisible.value = false
+  await completeValidation(false)
+  const task = tasks.value.find((x) => x.id === 'uvCache')
+  if (task) task.state = 'OK'
+}
+
+const openUrl = (url: string) => window.open(url, '_blank')
+
+const electronTasks: MaintenanceTask[] = [
```

Comment:
Let's store these tasks in a pinia store and define them in a hook like `coreCommandHooks`. `collect-i18n` should access these tasks and generate the `en/main.json`.

I am also fine if you expand the command definition and register some desktop + install time only commands to replace the task system here.